tls/extensions/s2n_client_psk.c

```diff
@@ -89,6 +89,46 @@ bool s2n_client_psk_should_send(struct s2n_connection *conn)
     return false;
 }
 
+/**
+ *= https://tools.ietf.org/rfc/rfc8446#section-4.2.11.1
+ *# The "obfuscated_ticket_age"
+ *# field of each PskIdentity contains an obfuscated version of the
+ *# ticket age formed by taking the age in milliseconds and adding the
+ *# "ticket_age_add" value that was included with the ticket (see
+ *# Section 4.6.1), modulo 2^32.
+*/
+static S2N_RESULT s2n_generate_obfuscated_ticket_age(struct s2n_psk *psk, uint64_t current_time, uint32_t *output)
+{
+    ENSURE_REF(psk);
+    ENSURE_MUT(output);
+
+    /**
+     *= https://tools.ietf.org/rfc/rfc8446#section-4.2.11
+     *# For identities
+     *# established externally, an obfuscated_ticket_age of 0 SHOULD be
+     *# used,
+     **/
+    if (psk->type == S2N_PSK_TYPE_EXTERNAL) {
+        *output = 0;
+        return S2N_RESULT_OK;
+    }
+
+    RESULT_ENSURE(current_time >= psk->ticket_issue_time, S2N_ERR_INTEGER_OVERFLOW);
+
+    /* Calculate ticket age */
+    uint64_t ticket_age_in_nanos = current_time - psk->ticket_issue_time;
+
+    /* Convert ticket age to milliseconds */
+    uint64_t ticket_age_in_millis = ticket_age_in_nanos / ONE_MILLISEC_IN_NANOS;
+    RESULT_ENSURE(ticket_age_in_millis <= UINT32_MAX, S2N_ERR_SAFETY);
+
```

Comment:
Is this a safety error because we've already enforced the "no older than a week" requirement when de-serializing the tickets into PSKs, so this should never happen?